### ISSUE TITLE
Implement RaP for iOS

### ISSUE THREAD

**Issue by terracoda** [phetsims/ratio-and-proportion#232]
I had an interview where the participant needed to use an iPhone and I realized the sim was not yet fully implemented to work on iOS.

This was not a huge surprise since I hadn't yet tested the sim on iOS.

Now is the time to start moving on that front.

The main issues currently are that when interacting with the individual hands there are no Responsive Descriptions firing - no objects responses and no context responses.

@zepumph, I am not sure if there is a process to get things going for iOS based on what is currently available, or if you need specific guidance from me about how the hand sliders and Both Hands are supposed to work on iOS. 

Both Hands will not work on iOS with VO on.

**Comment by terracoda** [comment]
@zepumph, please assign to me if you have questions.

**Comment by zepumph** [comment]
I am able to get context and object responses for the individual hand sliders on my iPad this morning. 7th gen iPad, ios 13.7. I'll update it and see if it still works.

**Comment by terracoda** [comment]
oh that's good.
Participant was using VO on iOS 13.

**Comment by terracoda** [comment]
In the interview, the participant moved the hands with a double tap and drag.

**Comment by terracoda** [comment]
We can talk on slack

**Comment by zepumph** [comment]
>the participant moved the hands with a double tap and drag.

That is not a supported interaction for sliders as far as I know. I can reproduce that swiping up/down works, but double tap and hold does not on native sliders with this snippet:

```html
<label for="he">Adjust range</label>
<input type="range" id="he"/>

<p id="ho">temp</p>
<script>
  document.getElementById( 'he' ).onchange = ( () => {
    document.getElementById( 'ho' ).innerText = document.getElementById( 'he' ).value;
  } );
</script>
```

I confirmed this same behavior on ios 14. Let me know what you would like to do from here.

**Comment by zepumph** [comment]
@terracoda showed me a video of a user interview where this was a problem. @emily-phet said in slack:

>@zepumph In this video Taliesin shared, the user is using mobile VO, and seems to be able to freely move the hands (left/right, up/down). Something strange seems to happen with the focus highlight when she is doing this (it disappears during hand motion, then appears again once she stops). She seems to only be able to hear description before moving the slider, and once she's done (which aligns with when the focus highlight is on the hand, so no description is heard when the focus highlight disappears). The fact that she's able to move the slider in un-slider-like ways, more like free gesture movement, makes me think that somehow her touch interaction was not intercepted by the screen reader, and instead was literally just moving the hand around as if the screen reader was not on. @terracoda suggested that this might have to do with a potential focus management issue. The way the participant described what she was doing was "double tap and hold" to activate the hand, and then moving to move the slider. We can't see her hands, so we're not sure exactly what was happening, or if her hand motion correlated directly with the movement of the slider.


>13:47
My guess is that ultimately this will end being some kind of issue with implementation, as the behavior we saw in this video is unlike anything @terracoda or I thought was possible with mobile VO interactions. It would be good if you could look into, and perhaps try it out yourself on the iPad, so you're familiar. If you're able to do that before tomorrow's meeting, we can discuss further then.

After watching the video, I understand more the issue here. We have never added support for this way of changing sliders (double tap and hold). The reason is that this operation sends a pointer event directly onto the hand. In this sense, the interaction becomes the exact same as mouse/touch. The solution here is to build out a way to get object responses sent (likely through aria-live) when using the ratio hand like this. arai-valuetext won't work because in this case, you don't have focus on the PDOM slider, but instead you are just using the visual/touch sim. Anything we do to fix this problem will add description that would be available when anyone used a mouse/touch interaction on the hands.  

I'm not sure we can build this out into AccessibleValueHandler. Perhaps something where you can use the accessibleValueHandler to register non-PDOM events so that the object responses can be spoken. 

We can discuss more tomorrow.

**Comment by zepumph** [comment]
I created https://github.com/phetsims/sun/issues/668, but from design meeting today, we decided that working on the double-tap and hold slider interaction on iOS shouldn't block the current publication track this sim is on. We can work on this when we start to implement self-voicing, a feature that @emily-phet mentioned is quite design-similar to this problem.

**Comment by zepumph** [comment]
We will close this issue, and handle this in the general sun issue
